src/main/java/net/minestom/server/collision/BlockCollision.java

```diff
@@ -375,12 +375,102 @@ static boolean checkBoundingBox(int blockX, int blockY, int blockZ,
                                     Vec entityVelocity, Pos entityPosition, BoundingBox boundingBox,
                                     Block.Getter getter, SweepResult finalResult) {
         // Don't step if chunk isn't loaded yet
-        final Block checkBlock = getter.getBlock(blockX, blockY, blockZ, Block.Getter.Condition.TYPE);
-        boolean hitBlock = false;
-        if (checkBlock.isSolid()) {
-            final Vec blockPos = new Vec(blockX, blockY, blockZ);
-            hitBlock = checkBlock.registry().collisionShape().intersectBoxSwept(entityPosition, entityVelocity, blockPos, boundingBox, finalResult);
+        final Block currentBlock = getter.getBlock(blockX, blockY, blockZ, Block.Getter.Condition.TYPE);
+        final Shape currentShape = currentBlock.registry().collisionShape();
+
+        final boolean currentCollidable = !currentShape.relativeEnd().isZero();
+        final boolean currentShort = currentShape.relativeEnd().y() < 0.5;
+
+        // only consider the block below if our current shape is sufficiently short
+        if(currentShort && shouldCheckLower(entityVelocity, entityPosition, blockX, blockY, blockZ)) {
+            // we need to check below for a tall block (fence, wall, ...)
+            final Vec belowPos = new Vec(blockX, blockY - 1, blockZ);
+            final Block belowBlock = getter.getBlock(belowPos, Block.Getter.Condition.TYPE);
+            final Shape belowShape = belowBlock.registry().collisionShape();
+
+            final Vec currentPos = new Vec(blockX, blockY, blockZ);
+            if(belowShape.relativeEnd().y() > 1) {
+                // we should always check both shapes, so no short-circuit here, to handle cases where the bounding box
+                // hits the current solid but misses the tall solid
+                return belowShape.intersectBoxSwept(entityPosition, entityVelocity, belowPos, boundingBox, finalResult)
+                        | (currentCollidable && currentShape.intersectBoxSwept(entityPosition, entityVelocity,
+                        currentPos, boundingBox, finalResult));
+            }
+            else {
```

Comment:
Could you inline `else` expressions? It is very confusing to read. Would also suggest using the `.editorconfig` file present in the repo root.

Reply:
Done on latest commit. I was using the .editorconfig from the beginning, but I manually set the line endings to CRLF before pushing since that's what the original files appeared to be using. BlockCollision should be on LF now.